### PR TITLE
feat(minifier): implement folding `charCodeAt` string fns

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

**Comment by camc314** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6475?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6475** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6475?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6436** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6436?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6435** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6435?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camc314 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs [hunk 1]
```diff
@@ -446,8 +479,10 @@ mod test {
         fold_same("x = 'abcde'.charCodeAt(5)"); // or x = (0/0)
         fold_same("x = 'abcde'.charCodeAt(-1)"); // or x = (0/0)
         fold_same("x = 'abcde'.charCodeAt(y)");
-        fold_same("x = 'abcde'.charCodeAt()"); // or x = 97
-        fold_same("x = 'abcde'.charCodeAt(0, ++z)"); // or (++z, 97)
+        // START: note the following test cases outputs differ from Google's
+        fold("x = 'abcde'.charCodeAt()", "x = 97");
+        fold("x = 'abcde'.charCodeAt(0, ++z)", "x = 97");
```
Comment (DonIsaac) [inline crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs:484]: This does not appear correct. `z++` is missing.

File: crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs [hunk 2]
```diff
@@ -155,6 +157,38 @@ impl PeepholeReplaceKnownMethods {
...
+                if unary_expr.operator == UnaryOperator::UnaryNegation =>
+            {
+                let Expression::NumericLiteral(numeric_lit) = &unary_expr.argument else {
+                    return None;
+                };
+                Some(-(numeric_lit.value))
+            }
+            None => None,
+            _ => return None,
+        };
```
Comment (7086cmd) [inline crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs:179]:
```rust
        let char_at_index = call_expr.arguments.first().and_then(|idx| ctx.get_number_value(idx))
```
Similar approches, you don't need to handle it manually.

**Review by 7086cmd — COMMENTED** [review]
I don't have enough time these days, so if things ready, call @Boshen to review and merge.

File: crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs (hunk 1 above)
Comment (camc314) [reply crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs:486]: fixed

File: crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs (hunk 2 above)
Comment (camc314) [reply crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs]: fixed

**Comment by Boshen** [comment]
### Merge activity

* **Oct 24, 10:35 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**